crud/request_test.go

```diff
@@ -70,27 +70,33 @@ var expectedOpts = map[string]interface{}{
 }
 
 type ValidSchemeResolver struct {
+	tarantool.SchemaResolver
 }
 
-func (*ValidSchemeResolver) ResolveSpaceIndex(s, i interface{}) (uint32, uint32, error) {
-	var spaceNo, indexNo uint32
+func (*ValidSchemeResolver) ResolveSpace(s interface{}) (uint32, error) {
+	var spaceNo uint32
```

Comment:
Yes, there are a couple of tests that need to resolve spaces and indexes:
https://github.com/tarantool/go-tarantool/blob/c564e6d75d10b1bfef276ebacc2eb7cf4acfdb24/crud/request_test.go#L469
https://github.com/tarantool/go-tarantool/blob/c564e6d75d10b1bfef276ebacc2eb7cf4acfdb24/crud/request_test.go#L631

https://github.com/tarantool/go-tarantool/blob/c564e6d75d10b1bfef276ebacc2eb7cf4acfdb24/crud/request_test.go#L98-L127

Reply:
The tests still work fine.
```patch
--- a/crud/request_test.go
+++ b/crud/request_test.go
@@ -3,7 +3,6 @@ package crud_test
 import (
     "bytes"
     "context"
-    "errors"
     "fmt"
     "testing"
 
@@ -12,7 +11,6 @@ import (
 
     "github.com/tarantool/go-tarantool/v2"
     "github.com/tarantool/go-tarantool/v2/crud"
-    "github.com/tarantool/go-tarantool/v2/test_helpers"
 )
 
 const invalidSpaceMsg = "invalid space"
@@ -70,45 +68,6 @@ var expectedOpts = map[string]interface{}{
     "timeout": timeout,
 }
 
-type ValidSchemeResolver struct {
-}
-
-func (*ValidSchemeResolver) ResolveSpace(s interface{}) (uint32, error) {
-    var spaceNo uint32
-    if s != nil {
-        spaceNo = uint32(s.(int))
-    } else {
-        spaceNo = defaultSpace
-    }
-    if spaceNo == invalidSpace {
-        return 0, errors.New(invalidSpaceMsg)
-    }
-    return spaceNo, nil
-}
-
-func (*ValidSchemeResolver) ResolveIndex(i interface{}, spaceNo uint32) (uint32, error) {
-    var indexNo uint32
-    if i != nil {
-        indexNo = uint32(i.(int))
-    } else {
-        indexNo = defaultIndex
-    }
-    if indexNo == invalidIndex {
-        return 0, errors.New(invalidIndexMsg)
-    }
-    return indexNo, nil
-}
-
-func (*ValidSchemeResolver) NamesUseSupported() bool {
-    isLess, err := test_helpers.IsTarantoolVersionLess(3, 0, 0)
-    if err != nil {
-        return false
-    }
-    return !isLess
-}
-
-var resolver ValidSchemeResolver
-
 func extractRequestBody(req tarantool.Request,
     resolver tarantool.SchemaResolver) ([]byte, error) {
     var reqBuf bytes.Buffer
@@ -125,12 +84,12 @@ func extractRequestBody(req tarantool.Request,
 func assertBodyEqual(t testing.TB, reference tarantool.Request, req tarantool.Request) {
     t.Helper()
 
-    reqBody, err := extractRequestBody(req, &resolver)
+    reqBody, err := extractRequestBody(req, nil)
     if err != nil {
         t.Fatalf("An unexpected Response.Body() error: %q", err.Error())
     }
 
-    refBody, err := extractRequestBody(reference, &resolver)
+    refBody, err := extractRequestBody(reference, nil)
     if err != nil {
         t.Fatalf("An unexpected Response.Body() error: %q", err.Error())
     }
```

Reply:
This is strange, because even on master branch we use resolver inside a `Body` call to call `res.ResolveSpaceIndex()`. And right now we are using the `resolver` in `newSpaceEncoder` and `newIndexEncoder` calls inside a `Body` call for `res.NamesUseSupported()` and `res.ResolveIndex()`.

Reply:
Yes, it is appears that we don't really use the resolver here. All because that all requests use the `NewCall17Request` type, that calls `tarantool.NewCallRequest`, for which we don't use resolver at all:
```
// Body fills an encoder with the call request body.
func (req *CallRequest) Body(res SchemaResolver, enc *msgpack.Encoder) error {
	args := req.args
	if args == nil {
		args = []interface{}{}
	}
	return fillCall(enc, req.function, args)
}
```

Reply:
Thanks, updated.